app/models/animal.rb

```diff
@@ -15,32 +19,5 @@ class Animal < ApplicationRecord
             :weight,
             :fun_fact,
             presence: true
+end
 
-  def predators_with_data
-    predator_names = self.predators.gsub(/\band\b/, '').split(',').map(&:strip).map(&:singularize)
-    Animal.where('name ILIKE ANY (ARRAY[?])', predator_names.map { |name| "%#{name}%" }).to_a
-  end
-
-  def self.find_animal(params)
-    find_by("name ILIKE ?", "%#{params[:name]}%")
-  end
-
-  def self.handle_predator_creation(animal)
-    predators = format_predators(animal)
-    predators.each do |predator_name|
-      next if Animal.where("name ILIKE ?", "%#{predator_name}%").exists?
-
-      animal_response = AnimalGateway.fetch_animal_data(predator_name)
-      photo_response = AnimalGateway.fetch_photo_data(predator_name)
-      if animal_response && photo_response
-        new_animal = AnimalDetail.new(animal_response, photo_response).as_json
-        Animal.create(new_animal)
-      end
-    end
-  end
-
-  def self.format_predators(animal)
-    animal.predators.gsub(/\band\b/, '').split(',').map(&:strip).map(&:singularize)
-  end
-
```

Comment:
I remember testing this for waaaay too long and am happy to see a refactor here. I assume the new tables and accompanying model classes are handling most of this business logic?